pkg/kubectl/cmd/cmd.go

```diff
@@ -303,6 +304,49 @@ func NewDefaultKubectlCommand() *cobra.Command {
 	return NewDefaultKubectlCommandWithArgs(NewDefaultPluginHandler(plugin.ValidPluginFilenamePrefixes), os.Args, os.Stdin, os.Stdout, os.Stderr)
 }
 
+func hasNoOptDefVal(name string, fs *pflag.FlagSet) bool {
+	flag := fs.Lookup(name)
+	if flag == nil {
+		return false
+	}
+	return flag.NoOptDefVal != ""
+}
+
+func shortHasNoOptDefVal(name string, fs *pflag.FlagSet) bool {
+	if len(name) == 0 {
+		return false
+	}
+
+	flag := fs.ShorthandLookup(name[:1])
+	if flag == nil {
+		return false
+	}
+	return flag.NoOptDefVal != ""
+}
+
+func appendPersistentFlagsAfterCommand(args []string, c *cobra.Command) []string {
+	flags := c.PersistentFlags()
+	for i := 0; i < len(args); i++ {
+		s := args[i]
+		switch {
+		case strings.HasPrefix(s, "--") && !strings.Contains(s, "=") && !hasNoOptDefVal(s[2:], flags):
```

Comment:
Trying to wrap my head around what this case is saying. I find the double-negative confusing `!hasNoOptDefVal`.

I think this case is matching an arg starting with `--`, without an `=` and without a default value?

Maybe rename `hasNoOptDefVal` so you don't have to negate it.